packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -61,6 +61,34 @@ class PlatformViewsRegistry {
 /// The `id` parameter is the platform view's unique identifier.
 typedef PlatformViewCreatedCallback = void Function(int id);
 
+/// Indicates the reason for a request to move keyboard navigation out of a platform view back into the widget tree.
+enum NavigationReason {
+  /// Non-directional
+  kProgrammatic,
+
+  /// Forward, e.g. Tab
+  kForward,
+
+  /// Backward, e.g. Shift + Tab
+  kBackward;
+
+  factory NavigationReason.fromInt(int index) {
```

Comment:
I would either add a comment as to why we're doing this manually, or, move this logic into `PlatformViewsService._onMethodCall`